common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java

```diff
@@ -332,25 +389,96 @@ public static Long parseLong(String value, int radix) {
+    private static int parseBinaryStringAsSignedInteger(String binaryString) {
+        if (binaryString.length() != 32) {
+            // Pad the binary string to 64 bits if it is not already
+            binaryString = String.format("%32s", binaryString).replace(' ', '0');
+        }
+
+        // If the MSB is 1, the number is negative in two's complement
+        if (binaryString.charAt(0) == '1') {
+            // Calculate the two's complement
+            String invertedBinaryString = invertBinaryString(binaryString);
+            int positiveValue = Integer.parseInt(invertedBinaryString, MIN_RADIX) + 1;
+            return -positiveValue;
+        } else {
+            return Integer.parseInt(binaryString, MIN_RADIX);
+        }
+    }
+    private static long parseBinaryStringAsSignedLong(String binaryString) {
+        if (binaryString.length() != 64) {
+            // Pad the binary string to 64 bits if it is not already
+            binaryString = String.format("%64s", binaryString).replace(' ', '0');
+        }
+
+        // If the MSB is 1, the number is negative in two's complement
+        if (binaryString.charAt(0) == '1') {
+            // Calculate the two's complement
+            String invertedBinaryString = invertBinaryString(binaryString);
+            long positiveValue = Long.parseLong(invertedBinaryString, MIN_RADIX) + 1;
+            return -positiveValue;
+        } else {
+            return Long.parseLong(binaryString, MIN_RADIX);
+        }
+    }
+
+    private static String invertBinaryString(String binaryString) {
+        StringBuilder invertedString = new StringBuilder();
+        for (char bit : binaryString.toCharArray()) {
+            invertedString.append(bit == '0' ? '1' : '0');
+        }
+        return invertedString.toString();
+    }
+
+    private static int getRadix10_16(String value) {
+        int radix = isDecimal(value) > 0 ? decRadix : isHexadecimal(value);
+        if (radix > 0) {
+            return radix;
+        } else {
+            throw new NumberFormatException("Value: \"" + value + "\" is not numeric or hexDecimal format!");
+        }
     }
 
     public static Float parseFloat(String value) {
-        if (value != null) {
+        return parseFloat(value, 0);
+    }
+
+    public static Float parseFloat(String value, int radix) {
+        if (StringUtils.isNotBlank(value)) {
+            String valueP = prepareNumberString(value);
+            int radixValue = isValidStringAndRadix(valueP, radix, value);
             try {
-                return Float.parseFloat(prepareNumberString(value));
+                if (radixValue == decRadix) {
+                    return Float.parseFloat(value);
+                } else {
+                    int bits = Integer.parseUnsignedInt(valueP, hexRadix);
+                    return Float.intBitsToFloat(bits);
+                }
             } catch (NumberFormatException e) {
+                throw new NumberFormatException(e.getMessage());
```

Comment:
Why do you catch and throw the same exception?

Reply:
Ok